app/frontend/wallet/account.ts

```diff
@@ -206,9 +208,11 @@ const Account = ({
     blockchainExplorer,
   })
 
-  async function ensureXpubIsExported() {
+  async function ensureXpubIsExported(): Promise<void> {
     // get first address to ensure that public key was exported
     await myAddresses.baseExtAddrManager._deriveAddress(0)
+    // in case the wallet is not shelley compatible we consider the
```

Comment:
mixing shelley compatibility and "usedness" seems like a potential great source of confusion :thinking:

do you see some way to reframe/refactor the logic to not have to "bend" the meaning of "used account"?

Reply:
I agree, I don't like this as well, that's why I added the comment, I think we are already discovering accounts only if the wallet is `ShelleyCompatible`. Och I just noticed, I misplaced this comment and it should be somewhere else..I ll fix it. Thnx.

Reply:
This probably deserves even more comment, since this logic is pretty crucial and the whole account discovery and xpub exporting depends on it.